src/main/java/com/google/javascript/gents/TypeScriptGenerator.java

```diff
@@ -225,8 +227,41 @@ public CodeGenerator getCodeGenerator(Format outputFormat, CodeConsumer cc) {
       }
     }
 
+    result.moduleRewriteLog = generateModuleRewriteLog(filesToConvert, modulePrePass);
     ((GentsErrorManager) compiler.getErrorManager()).doGenerateReport();
-    return sourceFileMap;
+    return result;
+  }
+
+  /**
+   * Generates a log file that maps how goog.modules/goog.provides where mapped to TS modules.
```

Comment:
+1 for using json (the gson lib is already a dependency). I think ease of future extensibility is a significant argument in favor.